metals/src/main/scala/scala/meta/internal/metals/watcher/ProjectFileWatcher.scala

```diff
@@ -172,17 +177,23 @@ object ProjectFileWatcher {
           watcher.register(root, Int.MaxValue)
         }
 
-        watcher
+        new FileWatcher {
+          override def start(): Unit = {}
+          override def cancel(): Unit = watcher.close
+        }
       } else {
         // Other OSes register all the files and directories individually
-        val watcher = initWatcher(watchFilter, watchEventQueue)
-
-        pathsToWatch.directories.foreach(
-          watcher.register(_, Int.MaxValue)
+        val close = SimpleFileWatcher.initWatcher(
+          watchFilter,
+          watchEventQueue,
+          pathsToWatch.files,
+          pathsToWatch.directories,
         )
-        pathsToWatch.files.foreach(watcher.register(_, -1))
 
-        watcher
+        new FileWatcher {
```

Comment:
Nit: Could we do something like this?
```
class SimpleFileWatcher(
  watchFilter: Path => Boolean,
  watchEventQueue: BlockingQueue[FileWatcherEvent],
  filesToWatch: Set[Path],
  directoriesToWatch: Set[Path],
)(implicit executionContext: ExecutionContext) extends FileWatcher {
  ...
}
```

It feels awkward to have a method that returns a closing function and to have to construct a `FileWatcher` from that.

Reply:
Yeah that works for me. I should have thought about that, I was just taking the shortest path from what I already had. :sweat_smile: